CI failing with missing jacoco file

### Description

Consistently, some tests with containers are failing with:

```
2025-08-19T22:35:21.3711284Z 22:35:21,321 INFO  [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#1]STDOUT: JAVA_OPTS already set in environment; overriding default settings with values: -javaagent:/opt/infinispan/server/lib/org.jacoco.agent-0.8.13-runtime.jar=output=file,destfile=/opt/jacoco.exec,append=true
2025-08-19T22:35:21.3714190Z 22:35:21,333 INFO  [t.l.i.s.p.jdbcstringbasedcachestoreit:latest] Container localhost/infinispan/server-o.i.s.p.jdbcstringbasedcachestoreit:latest started in PT0.13948968S
2025-08-19T22:35:21.3715892Z 22:35:21,333 INFO  [o.i.s.t.c.ContainerInfinispanServerDriver] Started container 1
2025-08-19T22:35:21.4743255Z 22:35:21,453 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: java.io.FileNotFoundException: /opt/jacoco.exec (Permission denied)
2025-08-19T22:35:21.4745193Z 22:35:21,454 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.io.FileOutputStream.open0(Native Method)
2025-08-19T22:35:21.4747020Z 22:35:21,455 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:289)
2025-08-19T22:35:21.4748792Z 22:35:21,455 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:230)
2025-08-19T22:35:21.4751387Z 22:35:21,456 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.output.FileOutput.openFile(FileOutput.java:75)
2025-08-19T22:35:21.4753552Z 22:35:21,458 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.output.FileOutput.startup(FileOutput.java:57)
2025-08-19T22:35:21.4755580Z 22:35:21,458 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.Agent.startup(Agent.java:129)
2025-08-19T22:35:21.4757525Z 22:35:21,458 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.Agent.getInstance(Agent.java:54)
2025-08-19T22:35:21.4759636Z 22:35:21,458 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.PreMain.premain(PreMain.java:47)
2025-08-19T22:35:21.4761821Z 22:35:21,459 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
2025-08-19T22:35:21.4763906Z 22:35:21,459 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
2025-08-19T22:35:21.4766200Z 22:35:21,459 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:560)
2025-08-19T22:35:21.4768715Z 22:35:21,459 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:572)
2025-08-19T22:35:21.4771312Z 22:35:21,460 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: Exception in thread "main" java.lang.reflect.InvocationTargetException
2025-08-19T22:35:21.4773825Z 22:35:21,460 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
2025-08-19T22:35:21.4776101Z 22:35:21,460 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
2025-08-19T22:35:21.4778250Z 22:35:21,460 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:560)
2025-08-19T22:35:21.4780913Z 22:35:21,461 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:572)
2025-08-19T22:35:21.4783170Z 22:35:21,461 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: Caused by: java.io.FileNotFoundException: /opt/jacoco.exec (Permission denied)
2025-08-19T22:35:21.4785038Z 22:35:21,461 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.io.FileOutputStream.open0(Native Method)
2025-08-19T22:35:21.4786843Z 22:35:21,462 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:289)
2025-08-19T22:35:21.4788897Z 22:35:21,462 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:230)
2025-08-19T22:35:21.4791018Z 22:35:21,462 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.output.FileOutput.openFile(FileOutput.java:75)
2025-08-19T22:35:21.4793093Z 22:35:21,463 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.output.FileOutput.startup(FileOutput.java:57)
2025-08-19T22:35:21.4795042Z 22:35:21,463 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.Agent.startup(Agent.java:129)
2025-08-19T22:35:21.4797142Z 22:35:21,464 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.Agent.getInstance(Agent.java:54)
2025-08-19T22:35:21.4799123Z 22:35:21,464 INFO  [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDOUT: FATAL ERROR in native method: processing of -javaagent failed, processJavaStart failed
2025-08-19T22:35:21.4801730Z 22:35:21,464 INFO  [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDOUT: Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
2025-08-19T22:35:21.4803655Z 22:35:21,465 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at org.jacoco.agent.rt.internal_0e20598.PreMain.premain(PreMain.java:47)
2025-08-19T22:35:21.4805815Z 22:35:21,465 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
2025-08-19T22:35:21.4807590Z 22:35:21,466 ERROR [o.i.CONTAINER] [o.i.s.p.JdbcStringBasedCacheStoreIT#0]STDERR: 	... 3 more
```

### Implementation ideas

_No response_

--- a/commons-test/src/main/java/org/infinispan/commons/test/ThreadLeakChecker.java
+++ b/commons-test/src/main/java/org/infinispan/commons/test/ThreadLeakChecker.java
@@ -31,7 +31,7 @@ import org.jboss.logging.Logger;
  * </p>
  * <ul>
  *    <li>Use -Dinfinispan.test.parallel.threads=3 (or even less) to narrow down source tests</li>
- *    <li>Set a conditional breakpoint in Thread.start with the name of the leaked thread</li>
+ *    <li>Set a conditional breakpoint in {@link Thread#start()} with the name of the leaked thread</li>
  *    <li>If the thread has the pattern of a particular component, set a conditional breakpoint in that component</li>
  * </ul>
  *
@@ -102,10 +102,12 @@ public class ThreadLeakChecker {
                       // Arjuna can start a Listener for recovery that is stopped in a shutdown hook
                       "|Listener:[0-9]" +
                       "|JVMCI-native CompilerThread[0-9]" +
-                      "|UNKNOWN[\\[\\]a-z]*" +
+                      "|JMX client heartbeat [0-9]" +
+                      "|RMI RenewClean" +
+                      "|RMI GC Daemon" +
+                      "|RMI Scheduler" +
                       ").*");
-   private static final String ARQUILLIAN_CONSOLE_CONSUMER =
-      "org.jboss.as.arquillian.container.CommonManagedDeployableContainer$ConsoleConsumer";
+   private static final Pattern ARQUILLIAN_CONSOLE_CONSUMER_REGEX = Pattern.compile("org\\.jboss(\\.as)?\\.arquillian\\.container[^$]+\\$ConsoleConsumer");
    private static final boolean ENABLED =
       "true".equalsIgnoreCase(System.getProperty("infinispan.test.checkThreadLeaks", "true"));
 
@@ -285,7 +287,7 @@ public class ThreadLeakChecker {
          // Special check for Arquillian, because it uses an unnamed thread to read from the container console
          StackTraceElement[] s = thread.getStackTrace();
          for (StackTraceElement ste : s) {
-            if (ste.getClassName().equals(ARQUILLIAN_CONSOLE_CONSUMER)) {
+            if (ARQUILLIAN_CONSOLE_CONSUMER_REGEX.matcher(ste.getClassName()).matches()) {
                return true;
             }
          }

